From 9743c7fc945de3e6b983434efb09f77d0fec1f7d Mon Sep 17 00:00:00 2001
From: Steve Kondik <shade@chemlab.org>
Date: Sat, 14 Jan 2012 17:04:22 -0800
Subject: [PATCH 2/6] adbd: Disable "adb root" by system property (2/3)

 * Require persist.sys.root_access=1 on non-eng debuggable builds

Change-Id: Iedab030e81ffb525fed64aed80cf0014f3e07073

adb: Fix compilation issue

Change-Id: Ia3284d5b8e428474726f64231f8142cf2474b17f

[mikeioannina] Adapt for O and rebrand prop

Change-Id: Ib880967a72296edfb05b80875fbe18136338a8c5
---
 adb/services.cpp | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/adb/services.cpp b/adb/services.cpp
index c6a8c5407..ae5ee7c44 100644
--- a/adb/services.cpp
+++ b/adb/services.cpp
@@ -84,6 +84,17 @@ void restart_root_service(int fd, void *cookie) {
             return;
         }
 
+        std::string root_access = android::base::GetProperty("persist.sys.root_access", "0");
+        std::string build_type = android::base::GetProperty("ro.build.type", "");
+        std::string lineage_version = android::base::GetProperty("ro.lineage.version", "");
+
+        if (!lineage_version.empty() && build_type != "eng" && (std::stoi(root_access) & 2) != 2) {
+            WriteFdExactly(fd, "root access is disabled by system setting - "
+                    "enable in Settings -> System -> Development options\n");
+            adb_close(fd);
+            return;
+        }
+
         android::base::SetProperty("service.adb.root", "1");
         WriteFdExactly(fd, "restarting adbd as root\n");
         adb_close(fd);
-- 
2.17.1

